app/Http/Controllers/MappingController.php

```diff
@@ -165,4 +170,56 @@ public function destroy(Webhook $webhook, Mapping $mapping)
             ]);
         }
     }
+
+    public function import(Webhook $webhook, MappingImportRequest $request)
+    {
+        DB::beginTransaction();
+        try {
+            $json = file_get_contents($request->file);
+            $getKeyValue = $this->mappingRepository->getKey($webhook);
+            $data = json_decode(file_get_contents($request->file), true);
+            $newData = [];
+            foreach($data as $key => $item) {
+                if (!in_array($item['key'], $getKeyValue)) {
+                    array_push($newData, array_merge($item, ["webhook_id" => $webhook->id]));
+                }
+            }
+
+            $this->mappingRepository->insert($newData);
+            DB::commit();
+            $request->session()->flash('messageSuccess', [
+                'status' => 'Upload file success',
+                'message' => 'This mapping successfully updaload file',
+            ]);
+
+        } catch (Exception $exception) {
+            DB::rollBack();
+            return response()->json([
+                'error' => true,
+                'message' => 'Upload file failed. Something went wrong',
+            ]);
+        }
+    }
+
+    public function exportJson(Webhook $webhook, Request $request)
+    {
+        try {
+            $mapping = $this->mappingRepository->getKeyAndValue($webhook);
+            $fileName = time() . '_datafile.json';
+            $newJsonString = json_encode($mapping, JSON_PRETTY_PRINT);
+            File::put(public_path('/json/'.$fileName),$newJsonString);
+            $request->session()->flash('messageSuccess', [
+                'status' => 'export file success',
+                'message' => 'This mapping successfully export file',
+            ]);
+
+            return response()->download((public_path('/json/'. $fileName)));
+
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingBeforeClose</br>**Severity**: error</br>**File**: app/Http/Controllers/MappingController.php</br>**Line**: 217</br>Blank line found at end of control structure